### ISSUE TITLE
[a11y] products page heading level issue

### ISSUE THREAD

**Issue by alexgibson** [mozilla/bedrock#15002]
## Description

The headings on the `/products/` page skip from `<h1>` straight to `<h3>`

### Steps to reproduce

https://www.mozilla.org/en-US/products/

### Expected result

The indivudal products should be `<h2>`

### Actual result

![image](https://github.com/user-attachments/assets/72b25220-6e9f-4624-ab77-da8a241a3358)

### Environment

N/A
